From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jakub Stasiak <jakub@stasiak.at>
Date: Sat, 7 Feb 2026 16:20:49 +0100
Subject: [PATCH] 8952: Improve the Namespace.getName() documentation

This is a bunch of more or less related improvements, hence all
of it bundled in one commit:

* Better terminology
* Cross-references where applicable
* Fixed getName(boolean) claiming that it returned "fully qualified name"
  unconditionally
---
 .../java/ghidra/program/model/symbol/Namespace.java | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/symbol/Namespace.java b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/symbol/Namespace.java
index 7f5b6a51ea..7ba1914b3f 100644
--- a/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/symbol/Namespace.java
+++ b/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/program/model/symbol/Namespace.java
@@ -81,15 +81,18 @@ public interface Namespace {
 	public boolean isExternal();
 
 	/**
-	 * Get the name of the symbol for this scope
-	 * @return the name of the symbol for this scope
+	 * Get the simple namespace name (without parent path).
+	 * See {@link #getName(boolean)} for the namespace-qualified variant.
+	 * @return the simple namespace name
 	 */
 	public String getName();
 
 	/**
-	 * Returns the fully qualified name
-	 * @param includeNamespacePath true to include the namespace in the returned name
-	 * @return the fully qualified name
+	 * Returns the namespace name, optionally prepended with the full parent namespace path.
+	 * @param includeNamespacePath true to include the namespace path
+	 * (using {@link #DELIMITER} as separator) in the returned name
+	 * @return the simple name if {@code includeNamespacePath} is false, or the full
+	 * namespace-qualified name (e.g., {@code "ClassA::InnerClass"}) if true
 	 */
 	public String getName(boolean includeNamespacePath);
 
-- 
2.45.1

